src/odemis/driver/test/xt_client_test.py

```diff
@@ -281,6 +283,158 @@ def test_set_beam_shift(self):
+        autostigmator_state = self.microscope.is_autostigmating(detector2ChannelName[detector])
+        self.assertEqual(autostigmator_state, True)
+        self.assertIsInstance(autostigmator_return, ProgressiveFuture)
+
+        autostigmator_return.cancel()
+        time.sleep(5.0)  # Give microscope/simulator the time to update the state
+        autostigmator_state = self.microscope.is_autostigmating(detector2ChannelName[detector])
+        self.assertEqual(autostigmator_state, False)
+
+        ## Test if autostigmating gives repeatable results and check time it takes to auto stigmate.
+
+        # Create sets holding stigmator values to check if the outcome is consistent
+        stigmator_values_x = []  # List holding only the stigmator values in x direction
+        stigmator_values_y = []  # List holding only the stigmator values in y direction
+
+        # Run apply_autostigmator multiple times and afterwards check if the outcome is the same for each run
+        for i in range(0, 5):
+            autostigmator_return = self.scanner.applyAutoStigmator(detector)
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            autostigmator_state = self.microscope.is_autostigmating(detector2ChannelName[detector])
+            self.assertEqual(autostigmator_state, True)
+            self.assertIsInstance(autostigmator_return, ProgressiveFuture)
+
+            starting_time = time.time()
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            while self.microscope.is_autostigmating(detector2ChannelName[detector]):
+                time.sleep(0.2)  # Wait until the autofocus is finished
+                if time.time() - starting_time > 30:
+                    break
+
+            # Check if the time to perform the autofocus is not to long
+            self.assertLess(time.time() - starting_time, 30, "Execution auto-stigmation is slow.")
+            # Line to inspect the execution time to update the expected time
+            print("Execution time was %s seconds " % (time.time() - starting_time))
+
+            stigmator_values_x.append(self.microscope.get_stigmator()[0])
+            stigmator_values_y.append(self.microscope.get_stigmator()[1])
+
+        # Check if all the stigmator values are within the required range each run
+        stigmator_range = self.microscope.stigmator_info()['range']
+        min_x_stig_values, max_x_stig_values = stigmator_range['x']
+        min_y_stig_values, max_y_stig_values = stigmator_range['y']
+        self.assertGreaterEqual(min(stigmator_values_x), min_x_stig_values)
+        self.assertLessEqual(max(stigmator_values_x), max_x_stig_values)
+        self.assertGreaterEqual(min(stigmator_values_y), min_y_stig_values)
+        self.assertLessEqual(max(stigmator_values_y), max_y_stig_values)
+
+        # TODO K.K. remove debug values after testing
```

Comment:
I don't think that should be merged.

Reply:
This test is currenlty not used because stigmator values cannot be applied yet. But indeed it is clean to remove this.